packages/flutter_tools/lib/src/commands/ios.dart

```diff
@@ -0,0 +1,164 @@
+  }
+
+  Future<List<int>> _fetchXcodeArchive() async {
+    print("Fetching the Xcode project archive from the cloud...");
+
+    HttpClient client = new HttpClient();
+    // TODO(chinmaygarde): Currently, engine releases are not tied to the release of the Xcode project
+    // archive. So use a "Current" tag. This will need to be replaced once released
+    // are in lockstep with the engine
+    HttpClientRequest request = await client.getUrl(_xcodeProjectUri("Current"));
+    HttpClientResponse response = await request.close();
+
+    if (response.statusCode != 200) {
+      throw new Exception(response.reasonPhrase);
+    }
+
+    BytesBuilder bytesBuilder = new BytesBuilder(copy: false);
+    await for (List<int> chunk in response) {
+      bytesBuilder.add(chunk);
+    }
+
+    return bytesBuilder.takeBytes();
+  }
+
+  Future<bool> _inflateXcodeArchive(String directory, List<int> archiveBytes) async {
+    print("Unzipping Xcode project to local directory...");
+
+    if (archiveBytes.length == 0) {
+      return false;
+    }
+
+    // We cannot use ArchiveFile because this archive contains file that are exectuable
+    // and there is currently no provision to modify file permissions during
+    // or after creation. See https://github.com/dart-lang/sdk/issues/15078
+    // So we depend on the platform to unzip the archive for us.
+
+    Directory tempDir = await Directory.systemTemp.create();
+    File tempFile = await new File(path.join(tempDir.path, "FlutterXcode.zip")).create();
+    IOSink writeSink = tempFile.openWrite();
+    writeSink.add(archiveBytes);
+    await writeSink.close();
+
+    ProcessResult result = await Process.run('/usr/bin/unzip', [tempFile.path, '-d', directory]);
+
+    // Cleanup the temp directory after unzipping
+    await Process.run('/bin/rm', ['rf', tempDir.path]);
+
+    if (result.exitCode != 0) {
+      return false;
+    }
+
+    Directory flutterDir = new Directory(path.join(directory, 'Flutter'));
+    bool flutterDirExists = await flutterDir.exists();
+    if (!flutterDirExists) {
+      return false;
+    }
+
+    // Move contents of the Flutter directory one level up
+    // There is no dart:io API to do this. See https://github.com/dart-lang/sdk/issues/8148
+
+    bool moveFailed = false;
+    List<FileSystemEntity> files = flutterDir.listSync();
+    for (FileSystemEntity file in files) {
+      ProcessResult result = await Process.run('/bin/mv', [file.path, directory]);
+      moveFailed = result.exitCode != 0;
+      if (moveFailed) {
+        break;
+      }
+    }
+
+    ProcessResult rmResult = await Process.run('/bin/rm', ["-rf", flutterDir.path]);
+    moveFailed = rmResult.exitCode != 0;
+
+    return !moveFailed;
+  }
+
+  Future<bool> _setupXcodeProjXcconfig(String filePath) async {
+    StringBuffer localsBuffer = new StringBuffer();
+
+    localsBuffer.writeln("// Generated. Do not edit or check into version control!!");
```

Comment:
Really?  Surprsied we don't want them to check it in.  Eventually we'll want that.